skyportal/handlers/api/source.py

```diff
@@ -606,13 +607,33 @@ async def get_sources(
             followup_request_subquery, Obj.id == followup_request_subquery.c.obj_id
         )
     if has_been_labelled:
-        labels_subquery = SourceLabel.select(session.user_or_token).subquery()
-        obj_query = obj_query.join(labels_subquery, Obj.id == labels_subquery.c.obj_id)
+        if current_user_labeller:
+            labels_subquery = (
+                SourceLabel.select(session.user_or_token)
+                .where(SourceLabel.labeller_id == user.id)
+                .subquery()
+            )
+            obj_query = obj_query.join(
+                labels_subquery, Obj.id == labels_subquery.c.obj_id
+            )
+        else:
+            labels_subquery = SourceLabel.select(session.user_or_token).subquery()
+            obj_query = obj_query.join(
+                labels_subquery, Obj.id == labels_subquery.c.obj_id
+            )
```

Comment:
Can we put only the `where` clause in the conditional and leave the rest of the code out of it?

Reply:
Agreed; change made!